core/templates/dev/head/domain/exploration/EditableExplorationBackendApiServiceSpec.js

```diff
@@ -0,0 +1,249 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for EditableExplorationBackendApiService.
+ */
+
+describe('Editable exploration backend API service', function() {
+  var EditableExplorationBackendApiService = null;
+  var ReadOnlyExplorationBackendApiService = null;
+  var sampleDataResults = null;
+  var $rootScope = null;
+  var $scope = null;
+  var $httpBackend = null;
+  var UndoRedoService = null;
+
+  beforeEach(module('oppia'));
+  beforeEach(module('oppia', GLOBALS.TRANSLATOR_PROVIDER_FOR_TESTS));
+
+  beforeEach(inject(function($injector) {
+    EditableExplorationBackendApiService = $injector.get(
+      'EditableExplorationBackendApiService');
+    ReadOnlyExplorationBackendApiService = $injector.get(
+      'ReadOnlyExplorationBackendApiService');
+    UndoRedoService = $injector.get('UndoRedoService');
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    $httpBackend = $injector.get('$httpBackend');
+
+    // Sample exploration object returnable from the backend
+    sampleDataResults = {
+      param_specs: {},
+      param_changes: [],
+      is_super_admin: false,
+      category: '',
+      rights: {
+        viewable_if_private: false,
+        status: 'private',
+        editor_names: [],
+        viewer_names: [],
+        cloned_from: null,
+        community_owned: false,
+        owner_names: ['jeaske']
+      },
+      is_admin: false,
+      objective: '',
+      states: {
+        Introduction: {
+          param_changes: [],
+          content: [{
+            type: 'text',
+            value: ''
+          }],
+          unresolved_answers: {},
+          interaction: {
+            customization_args: {},
+            fallbacks: [],
+            answer_groups: [],
+            default_outcome: {
+              param_changes: [],
+              dest: 'Introduction',
+              feedback: []
+            },
+            confirmed_unclassified_answers: [],
+            id: null
+          }
+        }
+      },
+      is_version_of_draft_valid: null,
+      version: 1,
+      user_email: 'test@example.com',
+      is_moderator: false,
+      language_code: 'en',
+      init_state_name: 'Introduction',
+      username: 'test',
+      tags: [],
+      title: '',
+      exploration_id: '0',
+      draft_changes: null,
+      profile_picture_data_url: '',
+      email_preferences: {
```

Comment:
I've removed a lot of this extra data, let me know if you think I have still included too much information.